src/lib/event.ts

```diff
@@ -0,0 +1,73 @@
+type memberof<obj> = obj[keyof obj];
+
+class Event<callback extends defined> {
+	_waitingThreads: thread[] = [];
+	_callbacks: callback[] = [];
+	name = "undefined";
+
+	constructor(name: string) {
+		this.name = name;
+	}
+
+	do(func: callback): this {
+		this._callbacks.push(func);
+		return this;
+	}
```

Comment:
How does one cancel/delete an event listener with precision?

Reply:
In general, most things referring to events here should be cancellable or bypassable

Reply:
I honestly have no idea, I'll get back to this when I'm not braindead

Reply:
Can we just delay this for later? I want to add in the logging system asap, and I want to get this done faster

Reply:
`.cancel()` on an event listener

Reply:
Will mark this as resolved, since I rewrote the events.